src/Dfe.EarlyYearsQualification.Web/Controllers/QualificationDetailsController.cs

```diff
@@ -141,31 +142,70 @@ public async Task<IActionResult> Index(string qualificationId)
+                                                               RatioRequirements.Level3RatioRequirementName,
+                                                               qualification);
+        model.RatioRequirements.RequirementsForLevel3 = await htmlRenderer.ToHtml(requirementsForLevel3);
+        
+        model.RatioRequirements.RequirementsHeadingForLevel3 =
+            GetRatioProperty<string>(additionalRequirementHeading, RatioRequirements.Level3RatioRequirementName, qualification);
 
         model.RatioRequirements.ApprovedForLevel6 =
-            CheckRatio(propertyToCheck, RatioRequirements.Level6RatioRequirementName, qualification);
+            GetRatioProperty<bool>(fullAndRelevantPropertyToCheck, RatioRequirements.Level6RatioRequirementName, qualification);
+        
+        var requirementsForLevel6 = GetRatioProperty<Document>(additionalRequirementDetailPropertyToCheck,
+                                                               RatioRequirements.Level6RatioRequirementName,
+                                                               qualification);
+        model.RatioRequirements.RequirementsForLevel6 = await htmlRenderer.ToHtml(requirementsForLevel6);
+        
+        model.RatioRequirements.RequirementsHeadingForLevel6 =
+            GetRatioProperty<string>(additionalRequirementHeading, RatioRequirements.Level6RatioRequirementName, qualification);
 
-        model.RatioRequirements.ApprovedForUnqualified = true;
+        model.RatioRequirements.ApprovedForUnqualified = 
+            GetRatioProperty<bool>(fullAndRelevantPropertyToCheck, RatioRequirements.UnqualifiedRatioRequirementName, qualification);
+        
+        var requirementsForUnqualified = GetRatioProperty<Document>(additionalRequirementDetailPropertyToCheck,
+                                                               RatioRequirements.UnqualifiedRatioRequirementName,
+                                                               qualification);
+        model.RatioRequirements.RequirementsForUnqualified = await htmlRenderer.ToHtml(requirementsForUnqualified);
+        
+        model.RatioRequirements.RequirementsHeadingForUnqualified =
+            GetRatioProperty<string>(additionalRequirementHeading, RatioRequirements.UnqualifiedRatioRequirementName, qualification);
     }
-
-    private bool CheckRatio(string propertyToCheck, string ratioName, Qualification qualification)
+    
+    private T GetRatioProperty<T>(string propertyToCheck, string ratioName, Qualification qualification)
     {
         try
         {
             var requirement = qualification.RatioRequirements!.Find(x => x.RatioRequirementName == ratioName);
 
-            return (bool)requirement!.GetType().GetProperty(propertyToCheck)!.GetValue(requirement, null)!;
+            return (T)requirement!.GetType().GetProperty(propertyToCheck)!.GetValue(requirement, null)!;
```

Comment:
The error logged in the `catch` block says "could not find property", but now it will also presumably throw an exception if the value can't be cast to the type `T`. Does this even matter? Should there be two `catch` blocks?

Reply:
I don't think it matters too much. However I have just pushed a minor change to log the exception so if this does happen we'll at least have a better idea as to why